harper-dictionary-parsing/src/lib.rs

```diff
@@ -1,24 +1,30 @@
 mod affix_replacement;
 mod attribute_list;
+pub mod char_string;
 mod error;
 mod expansion;
 mod matcher;
-mod word_list;
+pub mod span;
+pub mod word_list;
+pub mod word_metadata;
 
 pub use attribute_list::AttributeList;
 use attribute_list::HumanReadableAttributeList;
+pub use char_string::{CharString, CharStringExt};
 pub use error::Error;
+pub use span::Span;
+pub use word_metadata::WordMetadata;
 
 use self::word_list::parse_word_list;
 pub use self::word_list::MarkedWord;
 
 pub fn parse_default_word_list() -> Result<Vec<MarkedWord>, Error> {
-    parse_word_list(include_str!("../../../dictionary.dict"))
+    parse_word_list(include_str!("../dictionary.dict"))
 }
 
 pub fn parse_default_attribute_list() -> AttributeList {
     let human_readable: HumanReadableAttributeList =
```

Comment:
What are your thoughts on just moving all the dictionaries into this crate? That way, people (including us) can consume them on their own, without any of the rest of the Harper baggage.

Reply:
I just started trying this, but the issue is dependencies again. Because the build script depends on the parsing stuff, and the crate contains this, the `dictionary.fst` file has to be generated in `harper_core`, and since `FstDictionary` depends on `dictionary.fst` at compile-time, it has to be in `harper_core` as well.